sktime/transformations/series/window_summarizer.py

```diff
@@ -0,0 +1,357 @@
+__all__ = ["LaggedWindowSummarizer"]
+
+import pandas as pd
+from joblib import Parallel, delayed
+
+# from sktime.transformations.base import _PanelToTabularTransformer
+from sktime.transformations.base import BaseTransformer
+
+# List of native pandas rolling window function.
+# In the future  different engines for pandas will be investigated
+pd_rolling = [
+    "sum",
+    "mean",
+    "median",
+    "std",
+    "var",
+    "kurt",
+    "min",
+    "max",
+    "corr",
+    "cov",
+    "skew",
+    "sem",
+]
+
+
+def get_name_list(Z):
+    """Get names of pd.Series or pd.Dataframe."""
+    if isinstance(Z, pd.DataFrame):
+        Z_name = Z.columns.to_list()
+    else:
+        if Z.name is not None:
+            Z_name = [Z.name]
+        else:
+            Z_name = None
+    return Z_name
+
+
+def _window_feature(
+    Z,
+    name=None,
+    win_summarizer=None,
+    window=None,
+):
+    """Compute window features and lag.
+
+    Apply functions passed over a certain window
+    of past observations, e.g. the mean of a window of length 7 days, lagged by 14 days.
+
+    y: either pandas.core.groupby.generic.SeriesGroupBy
+        Object create by grouping across groupBy columns.
+    name : str, base string of the derived features
+           final name will also contain
+           window shift and window length.
+    win_summarizer: either str corresponding to native
+    implemented pandas function, currently
+            * "sum",
+            * "mean",
+            * "median",
+            * "std",
+            * "var",
+            * "kurt",
+            * "min",
+            * "max",
+            * "corr",
+            * "cov",
+            * "skew",
+            * "sem"
+         or function definition. See for the native functions also
+         https://pandas.pydata.org/docs/reference/window.html.
+    window: list of integers
+        Contains values for window shift and window length.
+    """
+    win = window[0]
+    shift = window[1] + 1
+
+    if win_summarizer in pd_rolling:
+        if isinstance(Z.index, pd.MultiIndex):
+            feat = getattr(Z.shift(shift).rolling(win), win_summarizer)()
+        else:
+            feat = pd.DataFrame(Z).apply(
+                lambda x: getattr(x.shift(shift).rolling(win), win_summarizer)()
+            )
+    else:
+        feat = Z.shift(shift)
+        if isinstance(Z.index, pd.MultiIndex) and callable(win_summarizer):
+            feat = feat.rolling(win).apply(win_summarizer, raw=True)
+        elif not isinstance(Z.index, pd.MultiIndex) and callable(win_summarizer):
+            feat = feat.apply(lambda x: x.rolling(win).apply(win_summarizer, raw=True))
+
+    if isinstance(feat, pd.Series):
+        feat = pd.DataFrame(feat)
+
+    feat.rename(
+        columns={
+            feat.columns[0]: name + "_" + "_".join([str(item) for item in window])
+        },
+        inplace=True,
+    )
+
+    return feat
+
+
+class LaggedWindowSummarizer(BaseTransformer):
```

Comment:
can we move this class up and the functions below? just a detail, but its more readable then